sqlmesh/core/plan/definition.py

```diff
@@ -162,14 +167,29 @@ def set_start(self, new_start: TimeLike) -> None:
     def end(self) -> TimeLike:
         """Returns the end of the plan or now."""
         if not self._end or not self.override_end:
-            if self.missing_intervals:
-                return max(
-                    end
-                    for intervals_per_model in self._missing_intervals.values()
+            if self._missing_intervals:
+                end, snapshot_version = max(
+                    (end, snapshot_version)
+                    for snapshot_version, intervals_per_model in self._missing_intervals.items()
                     for _, end in intervals_per_model
                 )
+                end_and_interval: t.Tuple[int, t.Optional[IntervalUnit]] = (-1, None)
```

Comment:
interval_and_units = [
(end, snapshot.model.interval_unit)
for snapshot in self.snapshots
for _, end in self._missing_intervals[snapshot.version_get_or_generate()]
]

... max(interval_and_units, key=lambda x: x[0])